examples/api/lib/widgets/inherited_model/inherited_model.0.dart

```diff
@@ -0,0 +1,169 @@
+    }
+    if (barSize != oldWidget.barSize && dependencies.contains('bar')) {
+      print('Rebuilding Bar Widget');
+      return  true;
+    }
+    return false;
+  }
+}
+
+class InheritedModelExample extends StatefulWidget {
+  const InheritedModelExample({super.key});
+
+  @override
+  State<InheritedModelExample> createState() => _InheritedModelExampleState();
+}
+
+class _InheritedModelExampleState extends State<InheritedModelExample> {
+  double _fooSize = 50.0;
+  double _barSize = 50.0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('InheritedModel Sample')),
+      body: Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: <Widget>[
+          Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              FooBarModel(
+                fooSize: _fooSize,
+                barSize: _barSize,
+                child: const FooWidget(),
+              ),
+              const SizedBox(height: 10.0),
+              ElevatedButton(
+                onPressed: () {
+                  setState(() {
+                    _fooSize = _fooSize == 50.0 ? 100.0 : 50.0;
+                  });
+                },
+                child: const Text('Resize Foo'),
+              ),
+
+            ],
+          ),
+          const SizedBox(width: 20.0),
+          Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              FooBarModel(
+                fooSize: _fooSize,
+                barSize: _barSize,
+                child: const BarWidget(),
+              ),
+              const SizedBox(height: 10.0),
+              ElevatedButton(
+                onPressed: () {
+                  setState(() {
+                    _barSize = _barSize == 50.0 ? 100.0 : 50.0;
+                  });
+                },
+                child: const Text('Resize Bar'),
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class FooWidget extends StatelessWidget {
+  const FooWidget({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final FooBarModel? fooBarModel = FooBarModel.of(context, 'foo');
+    final Color randomColor = Colors.primaries[Random().nextInt(Colors.primaries.length)];
+
+    return AnimatedContainer(
+      width: fooBarModel?.fooSize ?? 50.0,
+      height: fooBarModel?.fooSize ?? 50.0,
+      padding: const EdgeInsets.all(12.0),
+      color: randomColor,
+      duration: const Duration(seconds: 2),
+      curve: Curves.fastOutSlowIn,
+      alignment: Alignment.center,
+      child: const Text(
+        'Foo',
+        style: TextStyle(color: Colors.white),
+      ),
+    );
+  }
+}
+
+class BarWidget extends StatelessWidget {
+  const BarWidget({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    final FooBarModel? fooBarModel = FooBarModel.of(context, 'bar');
```

Comment:
Suggestion: Instead of making this return the model (from which you could accidentally access the fooSize without getting notified when it changes) maybe make this return the size of bar directly as a double. That way, it could also return a reasonable default if the FooBarModel doesn't exist. That way, the defaults are also defined in one central place.